packages/compass-query-bar/src/components/option-editor.tsx

```diff
@@ -148,26 +153,39 @@ const OptionEditor: React.FunctionComponent<OptionEditorProps> = ({
   }, [schemaFields, serverVersion]);
 
   const onFocus = () => {
-    if (hasAutofix && editorRef.current) {
-      if (editorRef.current.editorContents === '') {
+    if (hasAutofix && editorCurrentRef.current) {
+      if (editorCurrentRef.current.editorContents === '') {
         rafraf(() => {
-          editorRef.current?.applySnippet('\\{${}}');
+          editorCurrentRef.current?.applySnippet('\\{${}}');
         });
       }
     }
   };
 
   const onPaste = (event: React.ClipboardEvent<HTMLDivElement>) => {
-    if (hasAutofix && editorRef.current) {
-      const editorContents = editorRef.current.editorContents;
+    if (hasAutofix && editorCurrentRef.current) {
+      const editorContents = editorCurrentRef.current.editorContents;
       const snippet = lenientlyFixQuery(editorContents || '');
       if (snippet) {
-        editorRef.current.applySnippet(snippet);
+        editorCurrentRef.current.applySnippet(snippet);
         event.preventDefault();
       }
     }
   };
 
+  const onBlurEditor = useCallback(() => {
+    if (
+      !!editorCurrentRef.current &&
+      editorCurrentRef.current.editorContents !==
+        editorInitialValueRef.current &&
+      (editorInitialValueRef.current ||
+        editorCurrentRef.current.editorContents !== '{}')
+    ) {
+      track('Query Edited', { option_name: name });
```

Comment:
As this is implemented inside the option editor, this means that we will be tracking this event for filter, project, sort, collation, but not for skip, limit, and maxTimeMs, is this intentional? (Can't find this event in tracking plan sheet to double check myself)

Reply:
Thx for noticing this! We do need to track all find options.